src/transformers/image_transforms.py

```diff
@@ -858,3 +904,158 @@ def __call__(self, image: np.ndarray):
         # Same as in PyTorch, we assume incoming numpy images are in HWC format
         # c.f. https://github.com/pytorch/vision/blob/61d97f41bc209e1407dcfbd685d2ee2da9c1cdad/torchvision/transforms/functional.py#L154
         return torch.from_numpy(image.transpose(2, 0, 1)).contiguous()
+
+
+if is_torch_available():
+
+    class GroupByShape(torch.nn.Module):
+        """
+        Groups images by shape.
+        Returns a dictionary with the shape as key and a list of images with that shape as value,
+        and a dictionary with the index of the image in the original list as key and the shape and index in the grouped list as value.
+        """
+
+        def __init__(self):
+            super().__init__()
+
+        def forward(self, images: List["torch.Tensor"]):
+            grouped_images, grouped_images_index = group_images_by_shape(images)
+            return {"grouped_images": grouped_images, "grouped_images_index": grouped_images_index}
+
```

Comment:
It's fine for prototyping, but if we go further, we will need to redesign this. Hiding classes under `is_torch_available` looks weird.

Reply:
Agreed, but otherwise CI gets mad 😅